Removed refs to creating the 'router' and 'registry' service accounts

tpoitras opened the pull request:
The 'router' and 'registry' service accounts are now automatically created during installation (for Enterprise). And the image streams and quick start templates are pre-loaded during both the quick and advanced installation.

https://bugzilla.redhat.com/show_bug.cgi?id=1300503
Related to: https://bugzilla.redhat.com/show_bug.cgi?id=1266117

- Removed (made Origin-only using idef/endif) references to creating the 'router' and 'registry' service accounts (for enterprise), which are now automatically created during both quick and advanced installation.
- Moved "First Steps" content (for Enterprise) into new Admin guide topic as "Default Image Streams and Templates" (Enterprise-only)
- "First Steps" still exists but only for Origin
- Fixed all the inevitable broken links

tpoitras commented:
Tagging @bparees & @adellape for feedback on these changes (further to that e-mail I wrote openshift-sme list).

bparees commented on using_images/xpaas_images/a_mq.adoc:
```diff
@@ -42,7 +42,7 @@ xref:configuring-params[Application Template Parameters].
 xref:configuring-sti[Configuration Using S2I].
 
-ifdef::openshift-enterprise,openshift-origin[]
+ifdef::openshift-origin[]
```
why do the xpaas images not have an ifdef for enterprise instructions?

adellape replied:
@tpoitras The xPaaS topics are only included in the openshift-online,openshift-enterprise,openshift-dedicated distros in the build_cfg anyway, so any openshift-origin ifdefs within the topics won't matter. Really these could actually be openshift-enterprise ifdefs, that just say that they were installed by default during installation, and link to the new default_imagestreams_templates topic (and that way there's no link or mention of installation in the dedicated or online builds, too).

adellape commented on install_config/http_proxies.adoc:
```diff
@@ -112,8 +112,8 @@ NO_PROXY=master.hostname.example.com
-link:../install_config/install/first_steps.html#creating-instantapp-templates[example
-templates] available in OpenShift by default do not include settings for HTTP
+link:../install_config/default_imagestreams_templates.html#creating-instantapp-templates[example templates]
+available in OpenShift by default do not include settings for HTTP
```
Think this needs to be conditionalized between Origin/Enterprise.

adellape commented on install_config/install/deploy_router.adoc:
```diff
@@ -90,6 +94,7 @@ users:
 - system:serviceaccount:default:router
 ----
 ====
```
The manual editing of the SCC here could be updated to use the newer `add-scc-to-user` command if you wanted, as seen here:

https://docs.openshift.org/latest/admin_guide/manage_scc.html#add-an-scc-to-a-user-or-group

adellape commented on install_config/install/docker_registry.adoc:
```diff
@@ -121,9 +121,12 @@ container the request ultimately goes to.
+If you still want to use this option, create the registry and specify that it use the *registry* service account that was created during installation:
+endif::[]
+ifdef::openshift-origin[]
+If you still want to use this option:
```
Same here (below) wrt `add-scc-to-user`.

adellape commented on install_config/default_imagestreams_templates.adoc:
```diff
@@ -0,0 +1,222 @@
+link:../../architecture/core_concepts/builds_and_image_streams.html#image-streams[image streams] and link:../../architecture/core_concepts/templates.html[templates] to
+make it easy for developers to create new applications. By default, the
+link:quick_install.html[quick installation] and
+link:advanced_install.html[advanced installation] methods automatically create
```
The quick_install and advanced_install links are bad.

adellape replied:
Actually, all the links are bad because the file was moved up a level.

adellape commented on install_config/default_imagestreams_templates.adoc:
```diff
@@ -0,0 +1,222 @@
+$ oc create -f \
+    openshift-ansible/roles/openshift_examples/files/examples/v1.1/image-streams/image-streams-rhel7.json \
+    -n openshift
+----
```
All the JSON file paths for OSE should be updated per https://github.com/openshift/openshift-docs/issues/1454#issuecomment-173228118.

adellape replied:
See also https://github.com/openshift/openshift-docs/pull/1261#issuecomment-162090833.

tpoitras commented:
@adellape I have addressed your comments. However, I'm a bit unclear about the v1.1 thing for Origin. Right now the Origin docs are referring to the file paths with "v1.1" in them, but is that right? Would anyone in this current version of OpenShift Origin want to use the v1.0 path? 

Should I be explaining the difference between the versions here? If so, I'm unclear on why you'd use one version instead of the other.

Structurally, "First Steps" now just includes tagged content from "Default Image Streams and Templates", which is all conditionalised using ifdefs, especially where there are links (since default_imagestreams_templates.adoc and first_steps.adoc are at different levels in the file structure, all the links in first_steps were broken when I did the include::). Doing it this way still makes more sense to me than maintaining content in two different topics that are nearly-identical.

adellape commented:
@tpoitras Since we only maintain docs for "latest" Origin, I don't think we have a use for showing v1.0 paths.

Will check this out more closely tmw. :whale2:

adellape commented:
@tpoitras See https://github.com/tpoitras/openshift-docs/pull/1 for an add-on commit.

tpoitras commented:
Thanks, @adellape ! I have merged your PR into this one. :v: 

Moving this along to 2nd round of tech review now so we can merge it soon since it's a big move-around of stuff, and there's at least one BZ waiting on this. Tagging @brenton & @bparees to hopefully get a final tech review here on changes made by Alex and I.

Pretty build for Enterprise:

http://file.bne.redhat.com/tpoitras/2016/Feb24/openshift-enterprise/installautomation-BZ1300503/install_config/default_imagestreams_templates.html

Pretty build for Origin:

http://file.bne.redhat.com/tpoitras/2016/Feb24/openshift-origin/installautomation-BZ1300503/install_config/install/first_steps.html

tpoitras commented:
hang on, something is really messed up at the moment

bparees commented on architecture/core_concepts/templates.adoc:
```diff
@@ -87,12 +87,14 @@ application.
+link:../../install_config/install/first_steps.html#creating-instantapp-templates[define standard templates in the API]
+endif::[]
+ifdef::openshift-enterprise[]
+link:../../install_config/default_imagestreams_templates.html#creating-instantapp-templates[define standard templates in the API]
```
why do we have different locations for the same material between origin and OSE?

I also see that even the if-def:origin content in the "default_imagestreams_templates" file talks about this stuff, so why not link there?  (same for all the other cases below)

bparees commented on install_config/default_imagestreams_templates.adoc:
```diff
@@ -0,0 +1,420 @@
+link:advanced_install.html[advanced installation] method automatically creates
+these sets in the *openshift* project, which is a default project to which all
+users have view access.
+endif::[]
```
per comment above...why aren't we just linking here for the origin docs, instead of linking to first_steps?

bparees commented on install_config/default_imagestreams_templates.adoc:
```diff
@@ -0,0 +1,420 @@
+- link:../../using_images/s2i_images/perl.html[Perl]
+- link:../../using_images/s2i_images/php.html[PHP]
+- link:../../using_images/s2i_images/python.html[Python]
+- link:../../using_images/s2i_images/ruby.html[Ruby]
```
probably not worth fixing here, but we have a Wildfly image too that we really ought to get documented:
https://github.com/openshift/sti-wildfly/

bparees commented on install_config/default_imagestreams_templates.adoc:
```diff
@@ -0,0 +1,420 @@
+ifdef::openshift-origin[]
+- You must have cloned the
+https://github.com/openshift/openshift-ansible/tree/master/roles/openshift_examples/files[repository]
+that contains the default image streams and templates:
```
i thought the installer put them on the filesystem, why do they need to clone a repo to get them?
@sdodson

bparees replied:
i guess this instruction is origin specific, but at the top of this section we said the files were on the filesystem.

bparees commented on install_config/default_imagestreams_templates.adoc:
```diff
@@ -0,0 +1,420 @@
+link:../install_config/install/prerequisites.html#software-prerequisites[Software
+Prerequisites] for instructions.
+endif::[]
+- Define shell variables for the directories containing image streams and templates. This significantly shortens the commands in the following sections. To do this:
```
I guess the problem i'm having is separating where the "prereq" steps (the ones for getting the files in case you removed them from the filesystem) end, and the "regular" instructions resume.  maybe it's more obvious when rendered.  I would assume the prereq steps end just before the "define these shell variables" steps begin.

bparees commented on install_config/default_imagestreams_templates.adoc:
```diff
@@ -0,0 +1,420 @@
+database data will be lost if the database pod restarts for any reason.
+====
+endif::[]
+
```
I see no mention of what I'm calling "quickstart templates" that are listed here:
https://github.com/openshift/openshift-docs/pull/1627

pretty sure the installer lays down/installs those also.

Very similar to instantapp (in fact what is called instantapp here is really quickstart).

tpoitras replied:
I think the "quickstart vs instantapp" conversation is likely outside the scope of this PR

bparees replied:
that's fair, but we should still be talking about those other templates, even if we (incorrectly) refer to them as instantapps and come back later to fix the naming.

bparees commented on install_config/default_imagestreams_templates.adoc:
```diff
@@ -0,0 +1,420 @@
+If you used another method to install, or if the default image streams and
+templates are ever removed or changed, you can use the following instructions to
+create the default objects yourself. Otherwise, the following instructions are
+not necessary.
```
this whole section is a bit of a mess.

There are really 3 possible scenarios and they are being discussed in overlapping/unclear ways:
1) the content is registered with openshift.  life is good, nothing to do.
2) the content is not registered with openshift, but do exist on the filesystem (perhaps you accidentally unregistered them from your cluster).  you need to do most of the following steps, with the exception of the actual git clone (for origin) and rpm install (for OSE).
3) the content is neither on the local filesystem nor registered.  you need to do all of this.

reasons for (2) include:  someone unregistered them.  you never registered them in the first place.  you got updated versions (though in that case you still need to unregister the existing ones first, so you're back to the "someone unregistered them" path).

tpoitras replied:
@bparees @adellape 
Agreed this could be clearer. Perhaps we could approach it from another angle. "Run x command to see if you need to do all these things." 
What's the simplest way for an admin to tell if they need to do this or not?

bparees replied:
`oc get is -n openshift`
`oc get templates -n openshift`

will show what imagestreams and templates exist in the openshift namespace.

adellape replied:
That was my intent when adding the "You can check which default image streams and templates are currently available..." part in the Overview (web console + CLI methods).

bparees commented:
i'm not clear on the differences between the OSE and Origin install paths, but i'm bothered by the amount of if-def'ing we seem to have introduced as a result.  Having template/imagestream installation linked to two completely different locations depending on whether you're in origin or OSE docs is going to be a challenge to maintain going forward.  Is there no way to reconcile that back to a single path?

tpoitras commented:
@bparees You're absolutely right. Initially I thought that leaving the "First Steps" topic was important to have in its current location, but I'll move all the Origin content into the new "Default Image Streams and Templates" topic.

tpoitras commented:
@bparees 
made some more updates here.
no more "first steps" and both origin and enterprise are using the "default image stream and templates" topic (now simply named Image Streams and Templates). 

Speaking of that, @adellape , I dropped "Default" off since I was trying for a topic title that worked for both Online and OSE, I thought about "Provided Image Streams and Templates" but left it at this. You reckon it works?

bparees commented on architecture/core_concepts/templates.adoc:
```diff
@@ -87,12 +87,14 @@ application.
+link:../../install_config/imagestreams_templates.html#creating-instantapp-templates[define standard templates in the API]
+endif::[]
+ifdef::openshift-enterprise[]
+link:../../install_config/default_imagestreams_templates.html#creating-instantapp-templates[define standard templates in the API]
```
we can't give these files the same name?

adellape replied:
I imagine any remaining references to `default_imagestreams_templates` are just leftovers/typos at this point and are meant to be `imagestreams_templates`, since Files Changed no longer shows `default_imagestreams_templates`.

tpoitras replied:
Missed this one while running a search&replace script, thanks guys.

bparees commented on install_config/imagestreams_templates.adoc:
```diff
@@ -0,0 +1,318 @@
+----
+
+The JSON object definitions used to create these sets are installed on the file
+system of your master during
```
I think this is part of the confusion.  Are they installed on the filesystem, or are they registered in the cluster (or both)?  I think it's both.

But it's the "getting them registered in the cluster" that's important.  Having them on the filesystem is a pre-req to being able to register them in the cluster if you need to do so manually, but otherwise it's not interesting as a goal in itself.

adellape replied:
It's both if installed using the Quick or Advanced install. For OSE we know they must have used one of those methods, for Origin they can use Advanced (not Quick), but also may have installed using one of the 3 methods in https://docs.openshift.org/latest/getting_started/administrators.html, which would not lay down any example files.

bparees replied:
yeah, so what we really need to be walking them through is:

1) are they registered?  nothing to do
2) do you have the files?  no?  let's go get them
3) register the files

adellape replied:
+1

adellape commented on _build_cfg.yml:
```diff
@@ -179,6 +177,9 @@ Topics:
     File: master_node_configuration
+  - Name: Image Streams and Templates
+    File: imagestreams_templates
+    Distros: openshift-enterprise,openshift-origin
```
I don't think we need a Distros here since the whole book is already Distro'd for only Origin + Enterprise.

adellape commented:
> Speaking of that, @adellape , I dropped "Default" off since I was trying for a topic title that worked for both Online and OSE, I thought about "Provided Image Streams and Templates" but left it at this. You reckon it works?

I think we need something to distinguish this at a glance from being more than just a discussion about image streams and templates in general (which we already have other topics for). Maybe "Provided", but "Default" still seems fine to me.

adellape commented:
Was there a reason you thought "Default" wouldn't work for both Online and OSE?

tpoitras commented:
@adellape I thought that the word "default" implied that the image streams and templates were already loaded and ready for use. Would "loading" be technically accurate? As in changing the topic title to "Loading Image Streams and Templates" ?

tpoitras commented:
@adellape @bparees - updates as per your tech review

tpoitras commented:
@adellape this branch now has conflicts and I can't rebase from upstream since i pulled in your PR to mine. Not sure how to proceed?

tpoitras commented:
@adellape @bparees Does this pass tech review now?

adellape commented on install_config/install/deploy_router.adoc:
```diff
@@ -47,15 +47,19 @@ endif::[]
+Before deploying an OpenShift cluster, you must have a service account for the
+router. Starting in OpenShift Enterprise 3.1, a router
+link:../../architecture/core_concepts/projects_and_users.html[service account]
+is automatically created during a quick or advanced installation (previously, this required manual creation). This service account has permissions to a
```
```
a link:../../architecture/core_concepts/projects_and_users.html#users[service account] called *router* is automatically created in the *default* project during a quick or advanced installation
```

adellape commented on install_config/http_proxies.adoc:
```diff
@@ -107,13 +107,14 @@ NO_PROXY=master.hostname.example.com
 ----
 ====
 
+ifdef::openshift-enterprise[]
```
How come this section has an OSE-only ifdef around it?

adellape commented on install_config/http_proxies.adoc:
```diff
@@ -131,8 +132,9 @@ source:
 ====
 
-This is similar to the process for link:#git-repository-access[using proxies for
+This is similar to the process for link:../install_config/http_proxies.html#git-repository-access[using proxies for
```
This isn't needed when linking to an anchor in the same topic.

adellape commented on install_config/imagestreams_templates.adoc:
```diff
@@ -0,0 +1,326 @@
+endif::[]
+
+[[before-you-begin-istemps]]
+== Before You Begin
```
Do we need both a "Before You Begin" and a "Prerequisites"? I think we could leave out the header here and let the following (up to "Prerequisites") just be part of the "Overview", especially if it's framed as something to do before even bothering w/ the rest of  the topic.

bfallonf replied:
I'll agree that both a "Before you Begin" and "Prerequisites" probably isn't needed, but I think the placement should be before you read anything in the topic at all. How about an important admonition at the top of the Overview? We don't want to waste people's time be letting the read the Overview, then let them know that this topic might not be useful to them anyway.

tpoitras replied:
Changed "Before You Begin" into an admonition within the Overview. Was able to cut down on redundant explanation that way as well.

adellape commented on dev_guide/templates.adoc:
```diff
@@ -249,10 +249,9 @@ You can list the available default InstantApp templates with:
 $ oc get templates -n openshift
 ----
 
-ifdef::openshift-enterprise,openshift-origin[]
```
This ifdef is needed for the Online/Dedicated builds (which don't have the Install & Config guide).

adellape commented on using_images/db_images/mongodb.adoc:
```diff
@@ -284,10 +284,8 @@ link:../../architecture/core_concepts/pods_and_services.html#services[service].
 
 The MongoDB templates should have been registered in the default *openshift*
 project by your cluster administrator during the initial cluster setup.
-ifdef::openshift-enterprise,openshift-origin[]
```
Needed for Online/Dedicated guides.

adellape commented on using_images/db_images/mysql.adoc:
```diff
@@ -300,10 +300,8 @@ link:../../architecture/core_concepts/pods_and_services.html#services[service].
 
 The MySQL templates should have been registered in the default *openshift*
 project by your cluster administrator during the initial cluster setup.
-ifdef::openshift-enterprise,openshift-origin[]
```
Needed for Online/Dedicated guides.

adellape commented on using_images/db_images/postgresql.adoc:
```diff
@@ -285,10 +285,8 @@ link:../../architecture/core_concepts/pods_and_services.html#services[service].
 
 The PostgreSQL templates should have been registered in the default *openshift*
 project by your cluster administrator during the initial cluster setup.
-ifdef::openshift-enterprise,openshift-origin[]
```
Needed for Online/Dedicated guides.

adellape commented on using_images/other_images/jenkins.adoc:
```diff
@@ -91,10 +91,8 @@ define all the environment variables (password) with predefined defaults.
 OpenShift provides templates to make creating a new Jenkins service easy. The
 Jenkins templates should have been registered in the default *openshift* project
 by your cluster administrator during the initial cluster setup.
-ifdef::openshift-enterprise,openshift-origin[]
```
Needed for Online/Dedicated guides.

adellape commented on using_images/xpaas_images/data_grid.adoc:
```diff
@@ -207,14 +207,14 @@ Any configuration changes made using the JBoss Data Grid Management CLI on a run
 JBoss Data Grid instance inside the JBoss Data Grid xPaaS image] is different from the process you may be used to for a regular release of JBoss Data Grid.
 
-ifdef::openshift-enterprise,openshift-origin[]
+ifdef::openshift-origin[]
```
This should only be `openshift-enterprise`.

adellape commented on using_images/xpaas_images/decision_server.adoc:
```diff
@@ -79,14 +79,14 @@ ReleaseIds or Scanners, etc. are restricted.  The JMS endpoint currently does no
 both endpoints.
 
-ifdef::openshift-enterprise,openshift-origin[]
+ifdef::openshift-origin[]
```
This should only be `openshift-enterprise`.

adellape commented on using_images/xpaas_images/eap.adoc:
```diff
@@ -75,14 +75,14 @@ The following is a list of unsupported configurations specific to the JBoss EAP
 * Using MySQL in a scaled environment with XA distributed transactions is not recommended. For applications that support both scaling and XA distributed transactions, PostgreSQL is recommended instead.
 // This is based on https://issues.jboss.org/browse/CLOUD-56
 
-ifdef::openshift-enterprise,openshift-origin[]
```
This should only be `openshift-enterprise`.

adellape commented on using_images/xpaas_images/jws.adoc:
```diff
@@ -24,14 +24,14 @@ The location of `_JWS_HOME_/tomcat<version>/` inside a JBoss Web Server xPaaS im
 A major functionality difference compared to the regular release of JBoss Web Server is that there is no Apache HTTP Server in the OpenShift JBoss Web Server xPaaS images. All load balancing in OpenShift is handled by the OpenShift router, so there is no need for a load-balancing Apache HTTP Server with mod_cluster or mod_jk connectors.
 
-ifdef::openshift-enterprise,openshift-origin[]
+ifdef::openshift-origin[]
```
This should only be `openshift-enterprise`.

adellape commented:
> Would "loading" be technically accurate? As in changing the topic title to "Loading Image Streams and Templates" ?

@tpoitras I think this still loses that we're talking about a particular set of image streams and templates (if someone were looking at the topic name alone), not simply "how do I load is/templates", the latter of which we document in the [Dev Guide](https://docs.openshift.com/enterprise/latest/dev_guide/templates.html#uploading-a-template).

How about "Global Image Streams and Templates"? It both gives a name to the particular set(s) of is/templates we're talking about, but also doesn't suppose that you're going to necessarily be doing any manual actions.

Also, added comments inline on other stuff. Pardon some of the duplicated comments (in the `using_images` topics mainly); it was easier that way versus describing broadly here, for this topic versus that topic.

adellape commented:
@bparees Thoughts on "Global Image Streams and Templates" as a potential title?

bparees commented:
it's not my favorite but i can't say i have better suggestion at the moment

adellape commented:
That's about where I'm at as well.

bfallonf commented:
> > Would "loading" be technically accurate? As in changing the topic title to "Loading Image Streams and Templates" ?
> 
> @tpoitras I think this still loses that we're talking about a particular set of image streams and templates (if someone were looking at the topic name alone), not simply "how do I load is/templates", the latter of which we document in the [Dev Guide](https://docs.openshift.com/enterprise/latest/dev_guide/templates.html#uploading-a-template).

@adellape I feel this is a different type of thing than what's in the Dev Guide... 

> How about "Global Image Streams and Templates"? It both gives a name to the particular set(s) of is/templates we're talking about, but also doesn't suppose that you're going to necessarily be doing any manual actions.

Isn't this whole topic about performing a manual action? I'm not sure if "Global" is right in this context either.

It definitely needs a verb before the title, otherwise the reader is thinking this is a topic that's just telling them about the templates instead of how to load (or whatever) them. So maybe a combination of both? Maybe "Loading the Default Image Streams and Templates"? Or "Creating the Default Image Streams and Templates", seeing as how the subheadings use Create anyway?

I'm not sold on "Default", but they're the ones that are pre-loaded when you install using the quick or advanced install right? Usually, they'd be there by default...

adellape commented:
@bfallonf - 

> @adellape I feel this is a different type of thing than what's in the Dev Guide...

I just mean the Dev Guide shows how to "load" (create) a template in a project generically, and without a qualifier in the title it sounds like this topic is discussing image streams / templates generically.

> Isn't this whole topic about performing a manual action? I'm not sure if "Global" is right in this context either.

It would be in reference to these image streams and templates being available in the global "openshift" project. This topic is about those specific image streams & templates, how to get them (if they're missing), and where/how to load them (if they're missing).

> It definitely needs a verb before the title, otherwise the reader is thinking this is a topic that's just telling them about the templates instead of how to load (or whatever) them. So maybe a combination of both? Maybe "Loading the Default Image Streams and Templates"? Or "Creating the Default Image Streams and Templates", seeing as how the subheadings use Create anyway?

If they already have them loaded (which most people will), then it does (for them) end up being a topic that just explains the deal (i.e., we provide these, here's where they live, here's which ones you get per subscription). But the rest of the instructions are there in case they get in a bad way. While I agree adding "Loading" or "Creating" to the front may be correct in one sense, it may be a bit of a misnomer at a glance since most people won't need to manually load them.

> 
> I'm not sold on "Default", but they're the ones that are pre-loaded when you install using the quick or advanced install right? Usually, they'd be there by default...

Yes, that was the reason for the original naming.

bfallonf commented:
> I just mean the Dev Guide shows how to "load" (create) a template in a project generically, and without a qualifier in the title it sounds like this topic is discussing image streams / templates generically.

I think we're agreeing here.

> It would be in reference to these image streams and templates being available in the global "openshift" project. This topic is about those specific image streams & templates, how to get them (if they're missing), and where/how to load them (if they're missing).

I think this is bringing up a larger issue. Is there anywhere where we describe what the OpenShift and Default projects are actually doing? I didn't we talked about the OpenShift project before, but a quick search shows it's in the docs, but there's no description of its purpose. If it's described, then yes, I can understand why Global works in the title. If not, then we're just giving a name to something the reader might not particularly understand.

> > It definitely needs a verb before the title, otherwise the reader is thinking this is a topic that's just telling them about the templates instead of how to load (or whatever) them. So maybe a combination of both? Maybe "Loading the Default Image Streams and Templates"? Or "Creating the Default Image Streams and Templates", seeing as how the subheadings use Create anyway?
> 
> If they already have them loaded (which most people will), then it does (for them) end up being a topic that just explains the deal (i.e., we provide these, here's where they live, here's which ones you get per subscription). But the rest of the instructions are there in case they get in a bad way. While I agree adding "Loading" or "Creating" to the front may be correct in one sense, it may be a bit of a misnomer at a glance since most people won't need to manually load them.

If they already have them loaded, then there's no point for them to read this topic :P That's why I think a more descriptive title is best. Plus, you say so yourself above that this is about how and where to get them if it's missing. That's a task, and the point of this topic, if I'm not mistaken. There's only a para or two in the Overview that's explaining what they are.

adellape commented:
@bfallonf -

> I think this is bringing up a larger issue. Is there anywhere where we describe what the OpenShift and Default projects are actually doing? I didn't we talked about the OpenShift project before, but a quick search shows it's in the docs, but there's no description of its purpose. If it's described, then yes, I can understand why Global works in the title. If not, then we're just giving a name to something the reader might not particularly understand.

I think we should better describe (somewhere) both of those projects, and the other pre-fab projects, openshift-infra and management-infra (or at least verify whether they're worth mentioning currently).

> If they already have them loaded, then there's no point for them to read this topic :P That's why I think a more descriptive title is best. Plus, you say so yourself above that this is about how and where to get them if it's missing. That's a task, and the point of this topic, if I'm not mistaken. There's only a para or two in the Overview that's explaining what they are.

If they fully skip and don't read this topic, then there's no other admin-focused content that explains their existence (there's an obscure reference deep in the Arch guide and a better mention in the Dev Guide, both of which link to this topic). So this topic was currently pulling double duty (explaining them and, if needed, showing how to load/reload). Is there some better way to get the expository info to admins? Or do you think it is not needed?

adellape commented:
FWIW, I think ultimately sticking with "Default Image Streams and Templates" is still the best of the bunch.

bparees commented:
@adellape "Provided Image Streams and Templates" ?

bfallonf commented:
"Loading the Default Image Streams and Templates"? "Loading the Provided Image Streams and Templates"? We can let @tpoitras have the final say. It's his PR.

And @adellape I'll make a BZ about the openshift and default projects.

adellape commented:
> And @adellape I'll make a BZ about the openshift and default projects.

@bfallonf :+1:

tpoitras commented:
@adellape 
Updated based on comments from you and @bfallonf 
If there's nothing else, this should be ready to merge. :8ball:

tpoitras commented:
ping @adellape @bfallonf @tnguyen-rh @ahardin-rh for peer review.

This has been through 4 rounds of tech review and just needs a fresh set of eyes before it merges, thanks.

adellape commented:
@tpoitras Sorry for the delay. Will make sure to take a look as well in the morning. :eyes:

bparees commented on install_config/imagestreams_templates.adoc:
```diff
@@ -0,0 +1,321 @@
+
+[[creating-instantapp-templates]]
+
+== Creating InstantApp Templates
```
you will make @luciddreamz happy if you put a space in Instant App.

or at least less unhappy.

tpoitras replied:
I can do that, but does your comment below

> we are now distinguishing instant apps and quickstarts

trump this one?

bparees replied:
Creating Instant App and Quickstart Templates is probably the easiest solution.

bparees commented on install_config/imagestreams_templates.adoc:
```diff
@@ -0,0 +1,321 @@
+[[creating-instantapp-templates]]
+
+== Creating InstantApp Templates
+The InstantApp templates define a full set of objects for a running application.
```
same here

bparees commented on install_config/imagestreams_templates.adoc:
```diff
@@ -0,0 +1,321 @@
+to provide external access to the application.
+
+Some of the templates also define a database deployment and service so the
+application can perform database operations.
```
see: https://github.com/openshift/openshift-docs/pull/1627

we are now distinguishing instant apps and quickstarts, all of these are quickstarts except for the Jenkins templates.

tpoitras replied:
@bparees So are you suggesting the title of the section should be "Creating Quickstart Templates" instead? and remove references to InstantApp?

bparees replied:
should probably be "Creating Instant App and Quickstart Templates"

tpoitras commented:
Thanks, @bparees . Updated.

tpoitras commented:
ping @adellape @bfallonf @tnguyen-rh @ahardin-rh for peer review.

This has been through 5 rounds of tech review and just needs a fresh set of eyes before it merges, thanks.

Pretty rendered docs build for Enterprise:
http://file.bne.redhat.com/tpoitras/2016/Mar7/openshift-enterprise/installautomation-BZ1300503/install_config/imagestreams_templates.html

Pretty rendered docs build for Origin:
http://file.bne.redhat.com/tpoitras/2016/Mar7/openshift-origin/installautomation-BZ1300503/install_config/imagestreams_templates.html

bfallonf commented:
@tpoitras LGTM :100:

tnguyen-rh commented:
@tpoitras
One small nit: I see various instances of "you can direct your cluster admin to..." throughout (not just in this diff), and one "you can direct your developers to..." (carried over from the prior text).  Although the "you can direct your ROLE to..." construct is perfectly valid, i dislike it for several reasons:

- It is not succinct.
- It sets the reader apart from ROLE.
- It uses the possessive pronoun (which conflates relation and possession).

Granted, our docs are unlikely to be the seed of apocalyptic class warfare, but anyway, perhaps we can smooth things out a bit and consider replacing:

```asciidoc
You can direct your ROLE to link:URL[TITLE]
```

to

```asciidoc
ROLE can consult link:URL[TITLE]
```

or

```asciidoc
ROLE can link:URL[ACTION]
```

where `ACTION` is what is described if you follow `URL`.  This latter construct is a bit risky because it incurs higher maintenance awareness, but avoids the (potentially prententious-sounding) "consult"...

Otherwise (ignoring the nit), LGTM.  :smile:

tpoitras commented:
@tnguyen-rh that's valid but I think is a wider problem / out of scope for this BZ.

Merging.

tpoitras commented:
[rev_history]
|link:../install_config/imagestreams_templates.html[Loading the Default Image Streams and Templates]

|Moved and updated the First Steps topic to become the Loading the Default Image Streams and Templates topic.
